Fix broken logging hooks.

pasinskim opened the pull request:
Log messages are filtered out before reaching logging hooks if the log
level of logger is higher than the log level of given hook. This will
let logging hooks to filter all the messages according to their log levels.

Some refactoring done as well. Mostly removing functions which shouldn't
be exported and moving custom logging to one single place.

Signed-off-by: Marcin Pasinski marcin.pasinski@cfengine.com

pasinskim commented:
@kacf I don't think we are using this anywhere. Besides I think it is safer that our logger implementation will be the final one (aka you won't be able to create your own on top of Mender logger which is built on top of logrus).

ghost commented:
I kind of disagree, especially after learning how Go recommends that all data, including loggers, are passed in as arguments to the functions that use them. Especially in the server sections I think it's quite important that we keep this API, even if we haven't used it yet. And you are removing a feature which is already there and well tested, so I don't see it as being unsafe.

pasinskim commented:
@kacf Fixed! All previous exported functions are there untouched :)

kacf commented on log.go:
```diff
@@ -120,9 +132,9 @@ func PushModule(module string) {
-	self.moduleStack.Push(self.activeModule)
-	self.activeModule = module
+func (l *Logger) PushModule(module string) {
+	l.moduleStack.Push(Log.activeModule)
```
The module stack is logger local, `Log.` should be `l.`.

kacf commented on log.go:
```diff
@@ -135,12 +147,15 @@ func PopModule() {
-func (self *Logger) PopModule() {
-	self.activeModule = self.moduleStack.Pop().(string)
+func (l *Logger) PopModule() {
+	l.activeModule = Log.moduleStack.Pop().(string)
```
Same, `Log.` should be `l.`.

pasinskim committed:
Fix broken logging hooks.

Log messages are filtered out before reaching logging hooks if the log
level of logger is higher than the log level of given hook. This will
let logging hooks to filter all the messages according to their log levels.

Some refactoring done as well. Mostly removing functions which shouldn't
be exported and moving custom logging to one single place.

Signed-off-by: Marcin Pasinski <marcin.pasinski@cfengine.com>
